test/jdk/java/awt/Clipboard/CopyAnimatedGIFTest.java

```diff
@@ -0,0 +1,161 @@
+            (byte) 0x03, (byte) 0x00, (byte) 0x2c, (byte) 0x00, (byte) 0x00,
+            (byte) 0x00, (byte) 0x00, (byte) 0x04, (byte) 0x00, (byte) 0x04,
+            (byte) 0x00, (byte) 0x00, (byte) 0x02, (byte) 0x04, (byte) 0x94,
+            (byte) 0x8f, (byte) 0x29, (byte) 0x05, (byte) 0x00, (byte) 0x21,
+            (byte) 0xf9, (byte) 0x04, (byte) 0x01, (byte) 0x0a, (byte) 0x00,
+            (byte) 0x03, (byte) 0x00, (byte) 0x2c, (byte) 0x00, (byte) 0x00,
+            (byte) 0x00, (byte) 0x00, (byte) 0x04, (byte) 0x00, (byte) 0x04,
+            (byte) 0x00, (byte) 0x00, (byte) 0x02, (byte) 0x04, (byte) 0x8c,
+            (byte) 0x8f, (byte) 0x19, (byte) 0x05, (byte) 0x00, (byte) 0x3b
+    };
+
+    private void createGUI() {
+        img = Toolkit.getDefaultToolkit().createImage(imageData);
+
+        ImageCanvas canvas = new ImageCanvas(img);
+        canvas.setBackground(Color.BLUE);
+
+        Frame frame = new Frame("CopyAnimatedGIFTest");
+        frame.setSize(400, 200);
+        frame.add(canvas);
+        frame.setVisible(true);
+    }
+
+    private void copyImage() {
+        Clipboard sys = Toolkit.getDefaultToolkit().getSystemClipboard();
+        sys.setContents(new MyTransferable(img), null);
+
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        final CopyAnimatedGIFTest test = new CopyAnimatedGIFTest();
+
+        Robot robot = new Robot();
+
+        EventQueue.invokeAndWait(test::createGUI);
+
+        robot.waitForIdle();
+        robot.delay(1000);
+
+        EventQueue.invokeLater(() -> {
+            test.copyImage();
+            latch.countDown();
+        });
+
+        if (!latch.await(TIMEOUT, MILLISECONDS)) {
+            throw new RuntimeException("Image copying taking too long.");
+        }
+    }
+
+    private static class ImageCanvas extends Canvas {
+        private final Image img;
+        public ImageCanvas(Image img) {
+            this.img = img;
+        }
+
+        @Override
+        public void paint(Graphics g) {
+            g.drawImage(img, 0, 0, getSize().width, getSize().height, this);
+        }
+    }
+    private static class MyTransferable implements Transferable {
+        private final Image img;
+        private final DataFlavor[] flavors = {DataFlavor.imageFlavor};
+
+        public MyTransferable(Image img) {
+            this.img = img;
+        }
+
+        @Override
+        public DataFlavor[] getTransferDataFlavors() {
+            return flavors;
+        }
+
+        @Override
+        public boolean isDataFlavorSupported(DataFlavor flavor) {
+            return flavors[0].equals(flavor);
+        }
+
+        @Override
+        public Object getTransferData(DataFlavor flavor)
+                throws UnsupportedFlavorException {
+            if (!isDataFlavorSupported(flavor)) {
+                throw new UnsupportedFlavorException(flavor);
+            }
+            return img;
+        }
+    }
+
+}
+
```

Comment:
```suggestion
}
```
Now there are two blank lines in the end of the file for some reason. The thing is, GitHub and `git diff` don't usually show the blank line in the end of the file.

It's not too important, however, it may create noise when someone else updates the file and IDE removes the extra blank lines or adds ones if it's missing.